Lib/Database/MySqlQuery.php

```diff
@@ -0,0 +1,1247 @@
+        $value = $clause[1];
+        $oprt = '=';
+      } elseif (count($clause) == 3) {
+        $value = $clause[2];
+        $oprt = $clause[1];
+        if (!self::isOperatorValid($oprt, 'buildWhereClause'))
+          throw new RuntimeException(
+            'Arguments passed to `Query::'.$method.'()` aren\'t in the correct form'
+          );
+      }
+
+      if (is_array($value) && ($oprt == 'between' || $oprt == 'BETWEEN')) {
+        // Replace the value with a string that contains the values in the
+        // array, linked by 'AND'.
+        // If the array doesn't contains exactly 2 values, raise an exception.
+        if (count($value) != 2)
+          throw new RuntimeException(
+            'Arguments passed to `Query::'.$method.'()` aren\'t in the correct form'
+          );
+        try {
+          $value[0] = self::correctValue($value[0]);
+          $value[1] = self::correctValue($value[1]);
+        } catch (Exception $e) {
+          throw $e;
+        }
+        $value = $value[0].' AND '.$value[1];
+      } else {
+        try {
+          $value = self::correctValue($value, $method);
+        } catch (Exception $e) {
+          throw $e;
+        }
+      }
+      // Any other case (e.g. value is integer) is ok as it is.
+
+      $where .= $clause[0].' '.$oprt.' '.$value;
+    }
+
+    $this->where = $where;
+  }
+
+  /**
+   * Correct value.
+   *
+   * @since 1.0.0
+   * @access private
+   * @static
+   *
+   * @param mixed $value Scalar value.
+   * @param string $method Name of the method that called this one (directly or
+   * indirectly).
+   * @return mixed The correct value.
+   *
+   * @throws \RuntimeException if the value is an array.
+   */
+  private static function correctValue($value, $method) {
+    if (is_string($value)) {
+      // Put quotes around the value if it is a string.
+      $value = "'".$value."'";
+    } elseif (is_null($value)) {
+      // Replace the value with a string 'NULL' if it is null.
+      $value = 'NULL';
+    } elseif (is_bool($value)) {
+      // Replace the value with a string 'TRUE' or 'FALSE' if it is boolean.
+      $value = ($value == true) ? 'TRUE' : 'FALSE';
+    } elseif (!is_scalar($value)) {
+      throw new RuntimeException(
+        'Arguments passed to `Query::'.$method.'()` aren\'t in the correct form'
+      );
+    }
+    // Any other case (e.g. value is integer) is ok as it is.
+    return $value;
+  }
+
+  /**
+   * Verify the operator.
+   *
+   * Checks that the operator passed is a valid one.
+   *
+   * @since 1.0.0
+   * @access private
+   * @static
+   *
+   * @param string $operator Operator that needs to be verified.
+   * @param string $method Name of the method that called this one.
+   * @return boolean Whether the operator is valid or not.
+   */
+  private static function isOperatorValid($operator, $method) {
+    if ($method == 'join')
+      $valid_operators = array('=', '>', '>=', '<', '<=', '!=', 'like', 'LIKE');
```

Comment:
Put only uppercase versions of operators in valid operators arrays, and check against
```php
strtoupper($operator)
```